### ISSUE TITLE
[Filter panel] Website documentation

### ISSUE THREAD

**Issue by shixiedesign** [carbon-design-system/carbon-for-ibm-dotcom-website#961]
Craft 1. usage guidance and 2. design documentation for new component "Filter panel" content

**Comment by shixiedesign** [comment]
Some notes from design reviews:

### Filter panel 

- we don't control the amount of content in panel
- panel scrolls with page 
- panel sit inside the grid, doesn't hang right


- Keyboard interaction
		- Skip to main content / search results


Accordion vs "View all"
- going with Accordion, first category open on load
- Accordion is better for accessibility, allowing users skip through categories of filter 
- work on improving category label writing

Consider keeping filters per category limited so that users do not have to scroll/tab through more than _16_ filters at a time.

- keep categories with selected items open 
- if user closes it, no visual indicator, therefore need to have Search result tool bar to show current filters

Functions:
- Single select 
- Multi select
- 2 levels of hierarchy 
- Reset/clear per category

**Comment by shixiedesign** [comment]
More notes

Child container inside Filter panel 
- Position of the container is above filter groups
- Child in container is positioned in Modal on `md-sm` breakpoints

**Comment by oliviaflory** [comment]
All labels will the configurable for adopters, they will be responsible for translations (decided May 24 during

**Comment by oliviaflory** [comment]
@shixiedesign 

This particular component didn't fit into our normal documentation MVP template very well because of its complexity and customizable pieces. I tried to follow some of the other "rogue" categories we have on other special components like Lead space and Masthead to not deviate too far from the template. 


**Could use some feedback on what I have started:**
- any gaps in content/context that I have not covered
- Order of content, does the flow make sense?
   - Should I add an "Anatomy" section to point out what the different pieces are?
- Images are quick mocks-ups of what I thought could be helpful to illustrate what we are describing
   - Note: the mobile modal image in particular stinks, so I want to figure out a better way, suggestions?


I did rip the "Selection method" copy from Carbon with very minor tweaks 🙈   
I thought this was actually pretty important to cover.

![filter-panel--draft](https://user-images.githubusercontent.com/45692622/120554466-c5d65b80-c3c7-11eb-8553-f895d53fecc8.jpg)


Still need to figure out where to insert the documentation on the child container. Places where we could consider:
1. Under a "Variant", I didn't do this initially because it should just be a knob rather than a variant of the component. I don't think this is the best option.
2. Under "Behavior" and document the behavior of the Child container position in the two breakpoints, show two example images with desktop vs modal.
3. Under "Content guidance" and have a section dedicated to the Child container, with suggestions/recommendations of what type of content we suggest goes into the child container area.

This is what I have written so far for the Child content: 
>Use the tag group with filterable tags within the optional children area to help users see which filters they have selected. Suggested interaction: Deselecting filterable tags should also deselect the same filter within the filter panel.

---
Realizing this could be a bit annoying to give feedback on here...Happy to put it into markdown if that would be easier, the images are just such a pain 🤪  Will add my [Box note draft](https://ibm.box.com/s/87dvz8mbn0yx568kyykifn10m5sh6ph6) if anyone wanted to suggest edits ❤️
